### ISSUE TITLE
`@stylistic/declaration-block-trailing-semicolon` asks a Sass nested property for a semicolon it already carries

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#336]
Found by the review of the branch closing [#287](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/287) and [#335](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/335), and older than both. Until #335 the warning was swallowed by a fix that wrote nothing; with the fix declined it stands, and there is no way to write the file that clears it.

### What minimal example or steps are needed to reproduce the bug?

```scss
a { font: 12px { family: serif; }; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "customSyntax": "postcss-scss",
  "rules": {
    "@stylistic/declaration-block-trailing-semicolon": "always"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.scss" --config myconfig.json
```

### What did you expect to happen?

Nothing reported. The block ends with the semicolon the option asks for.

### What actually happened?

`1:14 Expected a trailing semicolon`, over a file that carries one. Writing a second semicolon does not clear it either, so the option is unsatisfiable for this shape.

### What is the cause?

The rule reads `parent.raws.semicolon`, which is the flag PostCSS hands the last node of a block. A Sass nested property closes with a brace, and the semicolon written behind that brace is parsed into the parent's own trailing raw instead: `a { font: 12px { family: serif; }; }` gives the rule `raws.semicolon` of `false` and `raws.after` of `"; "`.

That is the same split [#325](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/325) is about from the other side — there `never` clears the flag while the semicolon it means to take away lies in another raw. Both want the rule to read the semicolon the file holds rather than the flag alone.
